superset/views/core.py

```diff
@@ -2643,10 +2634,7 @@ def fetch_datasource_metadata(self):
             return json_error_response(DATASOURCE_MISSING_ERR)
 
         # Check permission for datasource
-        if not security_manager.datasource_access(datasource):
-            return json_error_response(
-                security_manager.get_datasource_access_error_msg(datasource),
-                link=security_manager.get_datasource_access_link(datasource))
+        self.check_datasource_permission(datasource)
```

Comment:
Same as mentioned above, this looks like it assumes that check_datasource_permission raises

Reply:
ditto